src/FondOfSpryker/Zed/DiscountCustomMessages/Business/Model/DiscountCustomMessagesReader.php

```diff
@@ -71,7 +72,31 @@ public function findDiscountCustomMessagesByIdDiscount(int $idDiscount): array
             return $this->createEmptyMessages();
         }
 
-        return $this->discountCustomMessagesMapper->mapTransfer($discountCustomMessagesEntities->getData());
+        return $this->discountCustomMessagesMapper->mapTransferCollection($discountCustomMessagesEntities->getData());
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\DiscountTransfer $discountTransfer
+     *
+     * @return \Generated\Shared\Transfer\DiscountCustomMessageTransfer|null
+     */
+    public function findCustomMessageByIdDiscountAndCurrentLocale(DiscountTransfer $discountTransfer): ?DiscountCustomMessageTransfer
+    {
+        if (!$discountTransfer->getIdDiscount()) {
+            return null;
+        }
+
+        $discountCustomMessagesEntity = $this->customMessagesQueryContainer
+            ->queryDiscountCustomMessageByIdDiscountAndIdLocale(
+                $discountTransfer->getIdDiscount(),
+                $this->localeFacade->getCurrentLocale()->getIdLocale()
+            );
```

Comment:
It's better to communicate with persistence in a repository without query container.